src/voting_nodes.rs

```diff
@@ -0,0 +1,214 @@
+//! The `vote_stage` nodes on the `last_id` of the bank at a regular cadence
+
+use byteorder::{LittleEndian, ReadBytesExt};
+use counter::Counter;
+use crdt::{Crdt, NodeInfo};
+use hash::Hash;
+use log::Level;
+use signature::Pubkey;
+use std::collections::HashMap;
+use std::io::Cursor;
+use std::sync::atomic::AtomicUsize;
+use std::sync::{Arc, RwLock};
+use timing::timestamp;
+use transaction::Vote;
+
+pub struct VotingNodes {
```

Comment:
Can you add documentation for this struct? I have a feeling there's a better name for this one. The current name might imply it does voting. Seems like something like `LiveNodes`, `NodeTracker`, or `NodeRegistrar` might be more fitting. Maybe even `BroadcastTableGenerator`? Let's start with a concise statement of its purpose and see if a clear, precise names falls out of that statement. I'm a bit concerned this module is serving two fairly unrelated purposes and curious if there's something we can do about it.